internal/core/metadata/utils.go

```diff
@@ -15,10 +15,11 @@ package metadata
 
 import (
 	"github.com/edgexfoundry/edgex-go/internal/core/data/errors"
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/config"
 )
 
-func checkMaxLimit(limit int) error {
-	if limit > Configuration.Service.MaxResultCount {
+func checkMaxLimit(limit int, config config.ConfigurationStruct) error {
```

Comment:
Should be passing config by reference instead of value.

Previously called out.  Please ensure all instances have been handled.

Reply:
Done. Following which, as usual, ran unit tests, ran BB tests 👍